ipympl/backend_nbagg.py

```diff
@@ -208,6 +208,44 @@ def send_binary(self, data):
     def new_timer(self, *args, **kwargs):
         return TimerTornado(*args, **kwargs)
 
+    def _repr_mimebundle_(self, **kwargs):
+        # now happens before the actual display call.
+        if hasattr(self, '_handle_displayed'):
+            self._handle_displayed(**kwargs)
+        plaintext = repr(self)
+        if len(plaintext) > 110:
+            plaintext = plaintext[:110] + '…'
+
+        buf = io.BytesIO()
+        self.figure.savefig(buf, format='png', dpi='figure')
+        data_url = b64encode(buf.getvalue()).decode('utf-8')
+
```

Comment:
@ianhi it seems that at this point in the code we still don't have the figure rendered (at least not the first time `display` gets called, so we need to call `update_display` later). It would be nice to not have to use `update_display` though, as it makes the widget flicker, and it wouldn't work properly from scripting tools like `nbconvert --execute`.

I've tried forcing rendering the figure there using `self.draw()` or `self.figure.draw(self.get_renderer())` here, but it doesn't work. Do you know how I could force the figure to be rendered **before** having received the `initialized` message from the front-end?

Reply:
In what way is it not working? Is the draw not happening or is savefig giving the wrong buffer or something?

Two potential avenues:
1. Maybe some of the trickery i proposed here: https://github.com/matplotlib/ipympl/issues/290#issuecomment-755377055
    - I think that the core of #290 is about when draws happen and the issue is back and forth with the frontend about `dpi` scaling.  
2. Can you try drawing and then instead of getting the buffer from savefig getting it similar to how it is done here: https://github.com/matplotlib/matplotli/blob/88ba86bcd41991fa81372b2f8c44710348195035/lib/matplotlib/backends/backend_webagg_core.py#L194-L196

Reply:
> In what way is it not working? 

The figure seems to have the right size but the buffer is completely empty, resulting in a white png.

> Can you try drawing and then instead of getting the buffer from savefig getting it similar to how it is done here:

IIRC I tried doing this already, with `renderer.buffer_rgba` but it was giving the same result. I can try again though.

> when draws happen and the issue is back and forth with the frontend about dpi scaling

That might be the issue...

Reply:
Actually this is expected to be blank...

When running the following code:
```python
fig = plt.figure()
plt.plot(np.sin(np.linspace(0, 20, 100)));
```
The first line will trigger a display of the widget even though we don't have any plot yet (which results in a blank mimetype). In the current state of the PR, the second line will trigger an `update_display`.

So I believe we can't get rid of `update_display` in the current state...